### PR TITLE
chore: handle date and time data type mappings for postgres and redshift

### PR DESCRIPTION
**Pull request by atzoum**
# Description

Handling mappings for some additional data types in postgres and redshift (including redshift data api driver):
- `DATE`
- `TIME WITHOUT TIMEZONE`
- `TIME WITH TIMEZONE`

Leaving legacy mappings intact

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: sqlconnect/internal/redshift/driver/rows.go [hunk 1]
```diff
@@ -81,21 +81,44 @@ func (rows *redshiftRows) Next(dest []driver.Value) error {
...
 					}
+				case "date":
+					t, err := time.Parse("2006-01-02", field.Value)
+					if err != nil {
+						dest[i] = field.Value
+					} else {
+						dest[i] = t.UTC().Format(time.RFC3339)
+					}
+				case "time", "time without time zone":
+					t, err := time.Parse("15:04:05", field.Value)
```
Comment (RanjeetMishra) [inline sqlconnect/internal/redshift/driver/rows.go:108]: Did not understand supporting time value without any date? Wondering why would anyone keep such value? 🤔

Comment (atzoum) [reply sqlconnect/internal/redshift/driver/rows.go:108]: It is important to expose consistent behaviour between the two drivers, `pq` & our custom `redshift-data` one. 

The `TIME` data type exists and this is how pq driver handles it. Whether this is useful or needs to be changed, this is a different issue altogether :)

File: sqlconnect/internal/postgres/testdata/column-mapping-test-seed.sql [hunk 2]
```diff
@@ -30,8 +33,8 @@ CREATE TABLE "{{.schema}}"."column_mappings_test" (
...
 VALUES
-    (1, 1,    1,    1,    1,    1,    1,    1,    1.1,  1.1,  1.1,  1.1,  1.1,  1.1,  'abc', 'abc', 'abc', 'abc', 'abc', 'abc', '2004-10-19 10:23:54+02', '2004-10-19 10:23:54', '2004-10-19 10:23:54+02', '2004-10-19 10:23:54+02', true,  true,  '{"a": 1}', '{"a": 1}'),
-    (2, 0,    0,    0,    0,    0,    0,    0,    0,    0,    0,    0,    0,    0,    '',    '',    '',    '',    '',    '',    '2004-10-19 10:23:54+02', '2004-10-19 10:23:54', '2004-10-19 10:23:54+02', '2004-10-19 10:23:54+02', false, false, '{}',       '{}'      ), 
-    (3, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL,  NULL,  NULL,  NULL,  NULL,  NULL,  NULL,                     NULL,                  NULL,                     NULL,                     NULL,  NULL,  NULL,       NULL      );
\ No newline at end of file
+    (1, 1,    1,    1,    1,    1,    1,    1,    1.1,  1.1,  1.1,  1.1,  1.1,  1.1,  'abc', 'abc', 'abc', 'abc', 'abc', 'abc', '2004-10-19', '10:23:54', '10:23:54+02', '2004-10-19 10:23:54+02', '2004-10-19 10:23:54', '2004-10-19 10:23:54+02', '2004-10-19 10:23:54+02', true,  true,  '{"a": 1}', '{"a": 1}'),
```
Comment (kuldeep0020) [inline sqlconnect/internal/postgres/testdata/column-mapping-test-seed.sql:38]: value for time and timetz looks off. 🤔

Comment (atzoum) [reply sqlconnect/internal/postgres/testdata/column-mapping-test-seed.sql:38]: Please elaborate, I don't understand what is the concern:
- `_time '10:23:54'` gets mapped to `"0000-01-01T10:23:54Z"`
- `_timetz '10:23:54+02'` gets mapped to `"0000-01-01T10:23:54+02:00"`

One is without a timezone and the other one is with a timezone, they are not equal

**Review by vyeshwanth — APPROVED** [review]

**Pull request merged**